src/debugger.c

```diff
@@ -146,18 +161,27 @@ int  DEBUGGetCurrentStatus(void) {
 
 	if (currentMode == DMODE_STEP) {							// Single step before
 		currentPC = pc;											// Update current PC
+		if (currentPC >= 0xA000) {								// Update the bank if we are in upper memory.
+			currentPCBank = currentPC < 0xC000 ? memory_get_ram_bank() : memory_get_rom_bank();
+		}
 		currentMode = DMODE_STOP;								// So now stop, as we've done it.
 	}
 
-	if (pc == breakPoint || pc == stepBreakPoint) {				// Hit a breakpoint.
+	if (hitBreakpoint(pc, breakPoint) || pc == stepBreakPoint) {// Hit a breakpoint.
```

Comment:
Does stepBreakPoint need to be changed from `int` to `struct breakpoint` as well? What happens if I do a step over in ROM bank 4 (BASIC stuff) and then chance upon the same PC in ROM bank 0 (kernal stuff)? Or ditto between two banks of cartridge memory?

Does the breakpoint logic need special case handling for memory addresses less than 0xa000? If I set a breakpoint in low-ram while the current ROM bank is 0, and hit it after a ROM bank switch to 4, does that disable the breakpoint in low-ram?

Reply:
Just committed a change that should address this. I had tried this earlier and it was acting buggy, but I think it was just that I hadn't set the bank at line 279.  One wonky thing I noticed that's happening is if you set a break point in lower memory, it shows the break point as FFFF0279, for example.  That's because I set the bank to -1 if it's in lower memory.  I'm not really sure how to address that..

Reply:
Never mind, I realized how right after I made that comment, duh 😅